modules/db/src/main/scala/gem/dao/ProgramDao.scala

```diff
@@ -45,19 +46,19 @@ object ProgramDao {
     pid.semesterOption.traverse(SemesterDao.canonicalize) *>
     Statements.insertNonstandardProgramIdSlice(pid).run
 
-  def selectBySubstring(pat: String, max: Int): ConnectionIO[List[Program[Nothing]]] =
+  def selectBySubstring(pat: String, max: Int): ConnectionIO[List[(Program.Id, String)]] =
     Statements.selectBySubstring(pat, max).to[List]
 
   /** Select a program by Id, without any Observation information. */
-  def selectFlat(pid: Program.Id): ConnectionIO[Option[Program[Nothing]]] =
+  def selectFlat(pid: Program.Id): ConnectionIO[Option[String]] =
```

Comment:
Yeah for now. Eventually it will include the science band and other crufty bits I guess. It will be good to add more structure here once we know which top-level queries we'll end up using.